### PR TITLE
CLN: assorted (non-pytables) cleanups

### PR DESCRIPTION
**Pull request by jbrockmendel**
includes a couple of whatsnew notes that fell through the cracks

### CODE REVIEW

File: pandas/tests/arrays/test_datetimelike.py [hunk 1]
```diff
@@ -40,8 +40,8 @@ def datetime_index(request):
     """
     freqstr = request.param
     # TODO: non-monotone indexes; NaTs, different start dates, timezones
-    pi = pd.date_range(start=pd.Timestamp("2000-01-01"), periods=100, freq=freqstr)
-    return pi
+    dti = pd.date_range(start=pd.Timestamp("2000-01-01"), periods=100, freq=freqstr)
+    return dti
```
Comment (gfyoung) [inline pandas/tests/arrays/test_datetimelike.py]: Why not just return `pd.date_range(...)`

**Review by gfyoung — COMMENTED** [review]
I think the changes look OK, but operationally speaking, I think it would have preferable to separate out some of these changes into separate PR's to facilitate review.

e.g. the `whatsnew` changes don't need to be held up by the `eval.py` changes.

**Comment by jbrockmendel** [comment]
@simonjayhawkins mypy is complaining about instantiating a class where we use abc.ABCMeta, thoughts?

**Comment by jbrockmendel** [comment]
@gfyoung broke pieces off into #30006, #30007, will revert those from this PR

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins mypy is complaining about instantiating a class where we use abc.ABCMeta, thoughts?

try this..

```diff
diff --git a/pandas/core/computation/engines.py b/pandas/core/computation/engines.py
index a4eaa897c..9b813bb4c 100644
--- a/pandas/core/computation/engines.py
+++ b/pandas/core/computation/engines.py
@@ -3,6 +3,7 @@ Engine classes for :func:`~pandas.eval`
 """
 
 import abc
+from typing import Dict, Type
 
 from pandas.core.computation.align import align_terms, reconstruct_object
 from pandas.core.computation.ops import _mathops, _reductions
@@ -139,4 +140,7 @@ class PythonEngine(AbstractEngine):
         pass
 
 
-_engines = {"numexpr": NumExprEngine, "python": PythonEngine}
+_engines: Dict[str, Type[AbstractEngine]] = {
+    "numexpr": NumExprEngine,
+    "python": PythonEngine,
+}
```

**Comment by jreback** [comment]
can you rebase

**Comment by jreback** [comment]
thanks @jbrockmendel if you can address @gfyoung comment in a followup

**Pull request merged**